agents/Group23/mcts.py

```diff
@@ -15,58 +16,52 @@ def __init__(self, colour: Colour, max_simulation_length: float = 2.5, custom_tr
         self.colour = colour  # Agent's colour
         self.max_simulation_length = max_simulation_length  # Length of a MCTS search in seconds
 
-    def _get_visit_count_distribution(self, node: TreeNode) -> list[list[int]]:
-        """Returns the visit count distribution for the children of the given node.
+    def run_simulation_with_process(self, root: TreeNode, colour: Colour,start_time: float) -> int:
```

Comment:
Looks like we've gone for processes as opposed to threads after all?

Reply:
Yeah, I started with threads, but they run one at a time in rotation instead of being truely parallel (ideal for IO tasks), which lead to a simular performance as before. But multi-processing is truely parallel so ideal for CPU heavy jobs.

https://medium.com/contentsquare-engineering-blog/multithreading-vs-multiprocessing-in-python-ece023ad55a

Reply:
Awesome, makes sense.